cpp/src/fil/fil.cu

```diff
@@ -625,30 +636,46 @@ void tl2fil_common(forest_params_t* params, const tl::Model& model,
     ASSERT(leaf_vec_size == model.num_output_group,
            "treelite model inconsistent");
     params->num_classes = leaf_vec_size;
-    params->leaf_payload_type = leaf_value_t::INT_CLASS_LABEL;
 
     ASSERT(tl_params->output_class,
            "output_class==true is required for multi-class models");
 
     ASSERT(
       pred_transform == "max_index" || pred_transform == "identity_multiclass",
       "only max_index and identity_multiclass values of pred_transform "
-      "are supported for multi-class models");
+      "are supported for multi-class models. provided: '%s'",
+      param.pred_transform);
 
+    params->leaf_payload_type = leaf_value_t::INT_CLASS_LABEL;
   } else {
-    ASSERT(pred_transform == "sigmoid" || pred_transform == "identity",
-           "only sigmoid and identity values of pred_transform "
-           "are supported for binary classification and regression models");
+    if (model.num_output_group > 1) {
+      params->num_classes = model.num_output_group;
+      ASSERT(tl_params->output_class,
+             "output_class==true is required for multi-class models");
+      ASSERT(pred_transform == "sigmoid" || pred_transform == "identity" ||
+               pred_transform == "max_index" ||
+               pred_transform == "multiclass_ova",
+             "only sigmoid, identity, max_index and multiclass_ova values of "
+             "pred_transform are supported for xgboost-style multi-class "
+             "classification models. provided: %s",
+             param.pred_transform);
+    } else {
+      params->num_classes = tl_params->output_class ? 2 : 1;
+      ASSERT(pred_transform == "sigmoid" || pred_transform == "identity",
+             "only sigmoid and identity values of pred_transform "
+             "are supported for binary classification and regression models. "
+             "provided: %s",
+             param.pred_transform);
+    }
     params->leaf_payload_type = leaf_value_t::FLOAT_SCALAR;
-    params->num_classes = 0;  // ignored
   }
 
   params->num_cols = model.num_feature;
 
   ASSERT(param.sigmoid_alpha == 1.0f, "sigmoid_alpha not supported");
   params->global_bias = param.global_bias;
   params->output = output_t::RAW;
-  if (tl_params->output_class) {
+  if (tl_params->output_class && pred_transform != "max_index") {
```

Comment:
What if it's multi-class, but `pred_transform != "max_index"`? Could you check for the number of classes instead and add a comment to make the code more clear?

Reply:
I would also need to check the leaf_algo_t, since the threshold doesn't apply for 1 or 2 classes with CATEGORICAL_LEAF.
Why is that clearer than not applying a threshold iff we use argmax?
I will certainly add a comment about these.